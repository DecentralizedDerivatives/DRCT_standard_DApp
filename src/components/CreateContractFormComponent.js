import React from 'react';
import { Field } from 'redux-form';
import { InputGroup } from 'reactstrap';
import InputNumber from './InputNumber';
import Select from './Select.js';
import { factories } from '../factoryProvider';

export const CreateContractFormComponent = ({
  handleSubmit,
  onSubmit,
  contractDates,
  onCurrencyChange
}) => {
  let factoryList = factories().reduce((obj, item) => {
    obj[item.address] = item.symbol + ' (' + item.multiplier + 'X multiplier)';
    return obj
  }, {})
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <InputGroup className="form-dropdown">
        <Field
          name="duration"
          label="Select Duration"
          component={Select}
          options={{
            'One Week': 'One Week'
          }}
        />
      </InputGroup>

      <InputGroup className="form-dropdown">
        <Field
          name="currency"
          label="Select Underlying"
          component={Select}
          options={factoryList}
          onChange={onCurrencyChange}
        />
      </InputGroup>
      <InputGroup className="form-dropdown">
        <Field
          name="startDate"
          label="Start Date"
          component={Select}
          options={contractDates}
        />
      </InputGroup>
          <p style={{margin:'5px'}}>Enter the amount of Ether for the total purchase of the contract below.</p>
      <InputGroup className="form-eth-input">
        <Field name="amount" addonLabel="ETH" step="0.01" component={InputNumber} />
      </InputGroup>

      <div className="form-submit-wrapper">
        <button type="submit" className="form-submit btn btn-primary">
          Create Contract
        </button>
      </div>
    </form>
  );
};

export default CreateContractFormComponent;
