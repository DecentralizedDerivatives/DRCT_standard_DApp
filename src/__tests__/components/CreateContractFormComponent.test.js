import CreateContractFormComponent from '../../components/CreateContractFormComponent';

function setup(overrides) {
  const store = initStore();
  const handleSubmit = jest.fn();
  const props = { store, handleSubmit, ...overrides };

  const wrapper = shallow(<CreateContractFormComponent {...props} />);

  return {
    wrapper,
  };
}

describe('<CreateContractFormComponent />', () => {
  it('renders the component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
