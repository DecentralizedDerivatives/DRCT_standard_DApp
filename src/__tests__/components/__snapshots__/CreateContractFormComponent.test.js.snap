// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateContractFormComponent /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContractFormComponent
    handleSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InputGroup
          className="form-dropdown"
          tag="div"
        >
          <Field
            component={[Function]}
            label="Select Duration"
            name="duration"
            options={
              Object {
                "One Week": "One Week",
              }
            }
          />
        </InputGroup>,
        <InputGroup
          className="form-dropdown"
          tag="div"
        >
          <Field
            component={[Function]}
            label="Select Underlying"
            name="currency"
            options={
              Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
              }
            }
          />
        </InputGroup>,
        <InputGroup
          className="form-dropdown"
          tag="div"
        >
          <Field
            component={[Function]}
            label="Start Date"
            name="startDate"
          />
        </InputGroup>,
        <p
          style={
            Object {
              "margin": "5px",
            }
          }
        >
          Enter the amount of Ether for the total purchase of the contract below.
        </p>,
        <InputGroup
          className="form-eth-input"
          tag="div"
        >
          <Field
            addonLabel="ETH"
            component={[Function]}
            name="amount"
          />
        </InputGroup>,
        <div
          className="form-submit-wrapper"
        >
          <button
            className="form-submit btn btn-primary"
            type="submit"
          >
            Create Contract
          </button>
        </div>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            component={[Function]}
            label="Select Duration"
            name="duration"
            options={
              Object {
                "One Week": "One Week",
              }
            }
          />,
          "className": "form-dropdown",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Select Duration",
            "name": "duration",
            "options": Object {
              "One Week": "One Week",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            component={[Function]}
            label="Select Underlying"
            name="currency"
            options={
              Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
              }
            }
          />,
          "className": "form-dropdown",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Select Underlying",
            "name": "currency",
            "onChange": undefined,
            "options": Object {
              "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
              "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            component={[Function]}
            label="Start Date"
            name="startDate"
          />,
          "className": "form-dropdown",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "label": "Start Date",
            "name": "startDate",
            "options": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter the amount of Ether for the total purchase of the contract below.",
          "style": Object {
            "margin": "5px",
          },
        },
        "ref": null,
        "rendered": "Enter the amount of Ether for the total purchase of the contract below.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            addonLabel="ETH"
            component={[Function]}
            name="amount"
          />,
          "className": "form-eth-input",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addonLabel": "ETH",
            "component": [Function],
            "name": "amount",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="form-submit btn btn-primary"
            type="submit"
          >
            Create Contract
          </button>,
          "className": "form-submit-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Contract",
            "className": "form-submit btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Create Contract",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InputGroup
            className="form-dropdown"
            tag="div"
          >
            <Field
              component={[Function]}
              label="Select Duration"
              name="duration"
              options={
                Object {
                  "One Week": "One Week",
                }
              }
            />
          </InputGroup>,
          <InputGroup
            className="form-dropdown"
            tag="div"
          >
            <Field
              component={[Function]}
              label="Select Underlying"
              name="currency"
              options={
                Object {
                  "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
                  "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
                }
              }
            />
          </InputGroup>,
          <InputGroup
            className="form-dropdown"
            tag="div"
          >
            <Field
              component={[Function]}
              label="Start Date"
              name="startDate"
            />
          </InputGroup>,
          <p
            style={
              Object {
                "margin": "5px",
              }
            }
          >
            Enter the amount of Ether for the total purchase of the contract below.
          </p>,
          <InputGroup
            className="form-eth-input"
            tag="div"
          >
            <Field
              addonLabel="ETH"
              component={[Function]}
              name="amount"
            />
          </InputGroup>,
          <div
            className="form-submit-wrapper"
          >
            <button
              className="form-submit btn btn-primary"
              type="submit"
            >
              Create Contract
            </button>
          </div>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Select Duration"
              name="duration"
              options={
                Object {
                  "One Week": "One Week",
                }
              }
            />,
            "className": "form-dropdown",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Select Duration",
              "name": "duration",
              "options": Object {
                "One Week": "One Week",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Select Underlying"
              name="currency"
              options={
                Object {
                  "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
                  "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
                }
              }
            />,
            "className": "form-dropdown",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Select Underlying",
              "name": "currency",
              "onChange": undefined,
              "options": Object {
                "0x92217550aba5912ba7eb70978871daf7d6bcc16d": "BTC/USD (1X multiplier)",
                "0xf55e6ce774cec3817467aed5f5a5769f006658d0": "ETH/USD (5X multiplier)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              component={[Function]}
              label="Start Date"
              name="startDate"
            />,
            "className": "form-dropdown",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "label": "Start Date",
              "name": "startDate",
              "options": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter the amount of Ether for the total purchase of the contract below.",
            "style": Object {
              "margin": "5px",
            },
          },
          "ref": null,
          "rendered": "Enter the amount of Ether for the total purchase of the contract below.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              addonLabel="ETH"
              component={[Function]}
              name="amount"
            />,
            "className": "form-eth-input",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addonLabel": "ETH",
              "component": [Function],
              "name": "amount",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="form-submit btn btn-primary"
              type="submit"
            >
              Create Contract
            </button>,
            "className": "form-submit-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Contract",
              "className": "form-submit btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Create Contract",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
